### ISSUE TITLE
peer dependencies for some plugins are broken

### ISSUE THREAD

**Issue by maribethb** [rachel-fenichel/blockly-samples#1120]
**Category**

Plugins

**Component**

block-extension-tooltip
dev-tools
serialize-disabled-interactions

**Describe the bug**

Some plugins have incompatible peer dependencies. see repro steps below. for people downloading these plugins and *not* using our lerna scripts, they could get an error like

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @blockly/dev-tools@3.1.1
npm ERR! Found: blockly@8.0.0
npm ERR! node_modules/blockly
npm ERR!   dev blockly@"^8.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer blockly@"3.20200123.0 - 7" from @blockly/theme-dark@2.0.7
npm ERR! node_modules/@blockly/theme-dark
npm ERR!   @blockly/theme-dark@"^2.0.7" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

this most likely only happens when someone has downloaded blockly-samples / is trying to run `npm install` in the plugin itself. It most likely does **not** affect someone who is just using the plugin as a dependency in their own project, because the incompatible dependencies are all in the devDependencies which aren't used except when developing the plugin. However it _could_ be a problem for people who are using dev-tools as a dependency in their own project or plugin and not using our lerna scripts.

**Cause**

When a plugin is published via lerna, the local packages using that plugin will be updated according to the normal npm rules about updating packages automatically. So if a minor version is created, plugins that use the `^1.0.0` syntax will automatically be updated to the new version. But if a major version is created, that could be a breaking change so it isn't updated automatically, and if we don't update them then our plugins remain using the old version of their dependencies, which require older versions of Blockly.

Why haven't we noticed the error?
We have the "forceLocal" flag enabled in our lerna config.
What this flag does is have lerna *always* use the local version of a plugin if it exists, even if it does not meet the version requirements listed in the plugin's package.json. That means if you install via lerna bootstrap, as we normally do, it will hide these errors from you.

What are the implications of removing that flag?
We have talked about removing this flag previously. We don't need it (I think it was originally added because it was thought that this flag was responsible for local linking at all, which is incorrect). If we removed it, then whenever a major version of a plugin is released and we don't update it in plugins that depend on it, then presumably CI would fail and we would know about this problem sooner and can fix it.

Can we update major version bumps automatically?
I haven't found anything in the lerna documentation to suggest this :/ however this problem is probably going to get worse if we use release-please, because it will be easier to create major-version bumps and not notice that we need to update other plugins in response, so we can check to see if release-please has any solutions for a monorepo in this position. A nice solution would be for a PR to be automatically opened like dependabot to update to the newest major version. It makes sense that it isn't bumped automatically because that could break things, since a major bump implies breaking changes. But it would be nice if there is an automated solution that doesn't rely on us remembering to update things.

Can we just use `>2.0.0` syntax or something so that the peerDependencies are always valid?
No, all versions need to have an upper bound. This is because we can't guarantee that a plugin will work with every future version of Blockly. We used to do this and it caused us problems because if a plugin isn't compatible with Blockly, then they may update Blockly and have no warning that the plugin is incompatible. Plugins need to explicitly list which version of Blockly they are compatible with.

**To Reproduce**

1. npm run clean:node
1. cd plugins/dev-tools
3. npm install (do not run npm install at the root level or run lerna bootstrap)
4. observe error

**Additional context**

TLDR: we need to update the dependencies for all plugins in the monorepo that depend on an older major version of other plugins, and then we need to remove the `forceLocal` flag from lerna config to prevent this happening again.
